lib/vssh/libssh2.c

```diff
@@ -958,6 +958,19 @@
   return result;
 }
 
+static int libssh2_key_exists(const char *filename)
+{
+  struct_stat sbuf;
+  if(!stat(filename, &sbuf)) {
+    FILE *fd = fopen(filename, FOPEN_READTEXT);
```

Comment:
Technically true, but the bigger issue is the double `fopen()`, not this. IMO.

Any suggestion how to deal with this?

Reply:
It seems OpenSSH does a `stat()`, then opens the file. Same this code did before this PR.

https://github.com/openssh/openssh-portable/blob/cbbbf76aa6cd54fce32eacce1300e7abcf9461d4/ssh-add.c#L1032

Reply:
Options are to ignore this warning, or to delete `stat()` to make it go away and pay the extra cycles.

I have a patch pending that will add 2 * 2 more `id_*` files to check, to match libssh.
Also worth noting that these checks are only performed if the use did not pass a private
key to (lib)curl explicitly via `--key` or `CURLOPT_SSH_PRIVATE_KEYFILE`. It also 
doesn't seem very safe to rely on the implicit keys, because anybody with FS or env
access can override them trivially.